geode-core/src/main/java/org/apache/geode/internal/util/ArgumentRedactor.java

```diff
@@ -16,98 +16,92 @@
+   * mark.<br>
+   * - Keys and values may be separated by an equals sign '=' or any number of spaces.<br>
+   * <br>
+   * Examples:<br>
+   * "--password=secret"<br>
+   * "--user me --password secret"<br>
+   * "-Dflag -Dkey=value"<br>
+   * "--classpath=."<br>
+   *
+   * @param line The argument input to be parsed
+   * @param permitFirstPairWithoutHyphen When true, prepends the line with a "-", which is later
+   *        removed. This allows the use on, e.g., "password=secret" rather than "--password=secret"
    *
-   * @return A printable string with redacted fields. E.g., "username=jdoe password=********"
+   * @return A redacted string that has sensitive information obscured.
    */
-  public static String redact(final Map<String, String> map) {
-    StringBuilder redacted = new StringBuilder();
-    for (Entry<String, String> entry : map.entrySet()) {
-      redacted.append(entry.getKey());
-      redacted.append("=");
-      redacted.append(redact(entry));
-      redacted.append(" ");
+  public static String redact(String line, boolean permitFirstPairWithoutHyphen) {
+
+    boolean wasPaddedWithHyphen = false;
+    if (!line.trim().startsWith("-") && permitFirstPairWithoutHyphen) {
+      line = "-" + line.trim();
+      wasPaddedWithHyphen = true;
     }
-    return redacted.toString().trim();
-  }
 
-  /**
-   * Returns the redacted value of the {@link Entry} if the key indicates redaction is necessary.
-   * Otherwise, value is returned, unchanged.
-   *
-   * @param entry A key/value pair
-   *
-   * @return The redacted string for value.
-   */
-  public static String redact(Entry<String, String> entry) {
-    return redact(entry.getKey(), entry.getValue());
+    // We capture the key, separator, and values separately, replacing only the value at print.
+    Matcher matcher = optionWithValuePattern.matcher(line);
+    while (matcher.find()) {
+      String option = matcher.group(1);
+      if (!containsTabooWord(option)) {
+        continue;
+      }
+
+      String separator = matcher.group(2);
+      String withRedaction = option + separator + redacted;
+      line = line.replace(matcher.group(), withRedaction);
+    }
+
+    if (wasPaddedWithHyphen) {
+      line = line.substring(1);
+    }
+    return line;
   }
 
   /**
-   * Parse a string to find key=value pairs and redact the values if necessary. If more than one
-   * key=value pair exists in the input, each pair must be preceded by a hyphen '-' to delineate the
-   * pairs. <br>
-   * Example:<br>
-   * Single value: "password=secret" or "--password=secret" Multiple values: "-Dflag -Dkey=value
-   * --classpath=."
-   *
-   * @param line The argument input to be parsed
-   * @return A redacted string that has sensitive information obscured.
+   * See {@link org.apache.geode.internal.util.ArgumentRedactor#redact(java.lang.String, boolean)}
+   * This alias permits the first key-value pair to be given without a leading hyphen, so that
```

Comment:
I think this javadoc would read better with the `See` link moved after the descriptive text.